packages/react-composites/src/composites/CallWithChatComposite/adapter/CallWithChatTypeAssertions.ts

```diff
@@ -0,0 +1,141 @@
+import { ChatThreadClientState } from '@internal/chat-stateful-client';
+import {
+  CallAdapterCallManagement,
+  CallAdapterClientState,
+  CallAdapterDeviceManagement,
+  CallAdapterUiState,
+  CallControlOptions
+} from '../../CallComposite';
+import { ChatAdapterThreadManagement, ChatAdapterUiState } from '../../ChatComposite';
+import { AdapterErrors } from '../../common/adapters';
+import { CallWithChatControlOptions } from '../CallWithChatComposite';
+import { CallWithChatAdapterUiState, CallWithChatClientState } from '../state/CallWithChatAdapterState';
+import { CallWithChatAdapterManagement } from './CallWithChatAdapter';
+/* @conditional-compile-remove(file-sharing) */
+import { FileUploadAdapter } from '../../ChatComposite';
+
+/// IMPORTANT
+///
+/// EACH INTERFACE DEFINED IN HERE IS THE PICK&OMIT VERSION OF THE FLATTENED INTERFACE WE EXPORT.
+/// WE ONLY EXPORT FLATTENED APIS FOR IMPROVED DEVELOPER EXPERIENCE. HOWEVER WE NEED TO ENSURE
+/// THAT THE SIGNATURES THAT CALLWITHCHAT USES MATCHES THE UNDERLYING SIGNATURES IN THE CALL
+/// AND CHAT INTERFACES. THUS HERE WE DEFINE INTERFACES PICK AND OMIT FROM THE CALL AND CHAT
+/// APIS AND ASSERT THAT THE FLATTENED APIS MATCH THE SIGNATURE.
+///
+/// THIS FILE WILL NEED KEPT UP TO DATE WHEN CHANGES TO THE FLATTENED INTERFACES ARE MADE.
+
+/// CallWithChatAdapterManagement
+
+interface CallWithChatAdapterManagementInternal
+  extends Pick<
+      CallAdapterCallManagement,
+      | 'startCamera'
+      | 'stopCamera'
+      | 'mute'
+      | 'unmute'
+      | 'startScreenShare'
+      | 'stopScreenShare'
+      | 'createStreamView'
+      | 'disposeStreamView'
+      | 'joinCall'
+      | 'leaveCall'
+      | 'startCall'
+    >,
+    Pick<
+      CallAdapterDeviceManagement,
+      | 'setCamera'
+      | 'setMicrophone'
+      | 'setSpeaker'
+      | 'askDevicePermission'
+      | 'queryCameras'
+      | 'queryMicrophones'
+      | 'querySpeakers'
+    >,
+    Pick<
+      ChatAdapterThreadManagement,
+      | 'fetchInitialData'
+      | 'sendMessage'
+      | 'sendReadReceipt'
+      | 'sendTypingIndicator'
+      | 'loadPreviousChatMessages'
+      | 'updateMessage'
+      | 'deleteMessage'
+    >,
+    /* @conditional-compile-remove(file-sharing) */
+    Pick<FileUploadAdapter, 'registerFileUploads' | 'clearFileUploads' | 'cancelFileUpload'> {}
+
+const CallWithChatAdapterManagementTypeAssertion = (
+  value: CallWithChatAdapterManagement
+): CallWithChatAdapterManagementInternal => value;
+
+const CallWithChatAdapterManagementRequiredTypeAssertion = (
+  value: Required<CallWithChatAdapterManagement>
+): Required<CallWithChatAdapterManagementInternal> => value;
+
+CallWithChatAdapterManagementTypeAssertion;
+CallWithChatAdapterManagementRequiredTypeAssertion;
+
+/// CallWithChatControlOptions
+
+interface CallWithChatControlOptionsInternal
+  extends Pick<CallControlOptions, 'cameraButton' | 'microphoneButton' | 'screenShareButton' | 'displayType'> {
+  chatButton?: boolean;
```

Comment:
All the extra fields here are unnecessary because we are only asserting one-way inclusion -- that the publicly exported API correctly extends the `*Internal` version.